packages/common/src/compiler/filtersCompiler.ts

```diff
@@ -121,39 +122,50 @@ export const renderNumberFilterSql = (
 export const renderDateFilterSql = (
     dimensionSql: string,
     filter: DateFilterRule,
+    adapterType: SupportedDbtAdapter,
     dateFormatter: (date: Date) => string = formatDate,
     startOfWeek: WeekDay | null | undefined = undefined,
 ): string => {
     const filterType = filter.operator;
+    const castValue = (value: string): string => {
```

Comment:
@vapolonio pushed a commit to simplify the changes in `renderDateFilterSql`. 
By having this `castValue` function we don't have to duplicate all the filters sql generation.
All unit tests are passing so it's generating the same SQL :)